remix-solidity/src/compiler/compiler-input.ts

```diff
@@ -25,9 +27,9 @@ module.exports = (sources, opts) => {
     o.language = opts.language
   }
   if (opts.language === 'Yul' && o.settings.optimizer.enabled) {
-    // details key is valid only for Yul
-    if (!o.settings.optimizer.details) o.settings.optimizer.details = {}
-    o.settings.optimizer.details['yul'] = true
+    if (!o.settings.optimizer['details']) 
+      o.settings.optimizer['details'] = {}
+    o.settings.optimizer['details']['yul'] = true
```

Comment:
Any reason not to do `o.settings.optimizer.details.yul` ?

Reply:
done